plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/TokenCredentialType.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.azure;
+
+// Type of token credentials that the plugin supports
+public enum TokenCredentialType {
+    MANAGED_IDENTITY,
```

Comment:
I don't understand these choices: what does it mean `NOT_APPLICABLE` ? Is the `MANAGED_IDENTITY` the only type?

Reply:
`NOT_APPLICABLE` is the default value when no token credential is selected/configured, which means a sas token or account key will be used for authentication instead of token credential such as ManageIdentityCredential. Happy to change it to a different name so that people could understand easily.

At the moment, `MANAGED_IDENTITY` is the only token credential type that is supported in this PR. I am assuming people might want to add on other token credential type such as WorkloadIdentity in the future.

Reply:
I don't think there is any use of NOT_APPLICABLE, if token is not managed / not configured - the property should not be provided at all.

Reply:
I have added a few comments on each choice, hope that would benefit future developers

Reply:
There is no need for comment, but this choice should not be presented: we should not expect anyone to configure: `token_credential_type=not_applicable`, it is just not helpful. It there is managed identity `token_credential_type`, should not be provided.

Reply:
Hello @reta, `NOT_APPLICABLE` is more like an internal representation of not using token credentials. That is an implementation limitation for having  `tokenCredentialType` of type `TokenCredentialType` in the `AzureStorageSettings` constructor because we cannot have a null/empty enum value. 

I might need to clarify that with the `NOT_APPLICABLE` implementation :
- Only when a user wants to use managed identity, they would configure by adding `azure.client.default.token_credential_type:managed_identity` to `opensearch.yml`. 
- The user would not have to add `azure.client.default.token_credential_type:not_applicable` to `opensearch.yml` if they do not want to use token credential over sas token/account key. 
- The implementation with the`NOT_APPLICABLE` ensures token_credential are not enable by default. If they want to use token credential, they have to configure in `opensearch.yml`.

In case, the explanation above is not convincing, I have committed a new implementation which change  `tokenCredentialType` to type `String` instead of `TokenCredentialType`  in the `AzureStorageSettings` constructor. That way, we can eliminate the usage of `NOT_APPLICABLE` completely. 

Welcome suggestions if others view this differently. Cheers.